services/frontend-service/src/ui/components/dropdown/dropdown.test.tsx

```diff
@@ -0,0 +1,134 @@
+
+Copyright 2021 freiheit.com*/
+
+import { DropdownSelect } from './dropdown';
+import { getByTestId, render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { UpdateOverview } from '../../utils/store';
+
+describe('Dropdown', () => {
+    const getNode = (overrides?: {}): JSX.Element | any => {
+        const defaultProps: any = {
+            children: null,
+        };
+        // given
+        return (
+            <div>
+                <MemoryRouter>
+                    <DropdownSelect {...defaultProps} {...overrides} />
+                </MemoryRouter>
+            </div>
+        );
+    };
+
+    const getWrapper = (
+        overrides?: {
+            className: string;
+            applications: any;
+            handleChange: (event: any) => void;
+            isEmpty: (arr: string[] | undefined) => boolean;
+            floatingLabel: string;
+            teams: string[];
+            selectedTeams: string[];
+        },
+        entries?: string[]
+    ) => render(getNode(overrides));
+
+    interface dataT {
+        name: string;
+        className: string;
+        handleChange: (event: any) => void;
+        isEmpty: (arr: string[] | undefined) => boolean;
+        floatingLabel: string;
+        teams: string[];
+        selectedTeams: string[];
+        applications: any;
+        expect: (container: HTMLElement) => void;
+    }
+
+    const data: dataT[] = [
+        {
+            name: 'Get label when no teams are selected',
+            className: 'top-app-bar-search-field',
+            handleChange: (event: any) => {},
+            isEmpty: (arr: string[] | undefined) => (arr ? arr.filter((val) => val !== '').length === 0 : true),
+            floatingLabel: 'Teams',
+            teams: ['example', 'bar'],
+            selectedTeams: [],
+            applications: {},
+            expect: (container) => {
+                const label = screen.getByTestId('teams-dropdown-label');
+                expect(label).toHaveAttribute('data-shrink', 'false');
+                expect(label).toHaveTextContent(/^Teams$/);
+            },
+        },
+        {
+            name: 'Get value after selecting a team',
+            className: 'top-app-bar-search-field',
+            handleChange: (event: any) => {},
+            isEmpty: (arr: string[] | undefined) => (arr ? arr.filter((val) => val !== '').length === 0 : true),
+            floatingLabel: 'Teams',
+            teams: ['example', 'bar'],
+            selectedTeams: ['example'],
+            applications: {},
+            expect: (container) => {
+                const label = screen.getByTestId('teams-dropdown-label');
+                expect(label).toHaveTextContent(/^Teams$/);
+                expect(
+                    getByTestId(container, 'teams-dropdown-select').getElementsByClassName('MuiSelect-select')[0]
+                ).toHaveTextContent(/^example$/);
```

Comment:
when there's so much code in the `expect` function (test array element), then it's a sign that this should be seperate test arrays.
I think this and the next array elements can be one test-array, but the first one (line 65) should be separate.

Reply:
I've separated the expect functions in what I understand to be test arrays! One to test the label, one to test if the label shrinks appropriately, and another to check if the dropdown text has the selected teams.